Added webpack 2 example

mauricedb opened the pull request:
Added a Webpack 2 configuration example as the syntax is slightly different with `rules` instead of `loaders`.

TheLarkInn commented on pages/Integrating with Build Tools.md:
```diff
@@ -167,6 +167,27 @@ module.exports = {
+    module: {
+        rules: [
+            // all files with a '.ts' or '.tsx' extension will be handled by 'ts-loader'
+            { test: /\.tsx?$/, loader: "ts-loader", exclude: /node_modules/ }
```
Please change loader to use

DanielRosenwasser replied:
For the record, I think @TheLarkInn is referring to https://webpack.js.org/configuration/module/#rule-use

TheLarkInn commented on pages/Integrating with Build Tools.md:
```diff
@@ -167,6 +167,27 @@ module.exports = {
+        filename: "bundle.js"
+    },
+    resolve: {
+        extensions: [".tsx", ".ts", ".js"]
```
This is an incorrect "appending" from the default values. See webpack.js.org/configuration/resolve

DanielRosenwasser commented on pages/Integrating with Build Tools.md:
````diff
@@ -167,6 +167,27 @@ module.exports = {
 }
 ```
 
+### Basic webpack.config.js when using Webpack 2
````
Can you swap sections for Webpack 1 & 2?

mauricedb commented:
@TheLarkInn and @DanielRosenwasser 

I have updated the PR with the requested changes. Please let me know if this is what you were looking for.

Also rebased the PR on master.

Maurice

mauricedb commented:
@DanielRosenwasser Is this ever going to get merged? The docs still only show the Webpack 1  configuration and we are at version 4 already.
